app/Http/Controllers/ApiController.php

```diff
@@ -79,42 +84,35 @@ public function renderPlaylistSongs(SpotifyWebAPI\SpotifyWebAPI $api)
         $api->setAccessToken(session()->get('access_token'));
         $songs = $api->getPlaylistTracks(request()->playlist_id)->items;
 
-        return view('baskets.import', [
+        return view('books.import', [
             'songs' => $songs
 
         ]);
     }
-
-    public function importPlaylist(SpotifyWebAPI\SpotifyWebAPI $api, Request $request)
+    public function importPlaylist()
     {
-        $api->setAccessToken(session()->get('access_token'));
-        foreach (request()->track_ids as $track_id) {
-            $this->getSongData($api, $track_id, request()->basket);
-        };
-
-        return redirect()->route('baskets.show', ['basket' => session()->get('basket')]);
-    }
-
-    private function getSongData(SpotifyWebAPI\SpotifyWebAPI $api, string $track_id)
-    {
-        $track = $api->getTrack($track_id);
-
-        $song_data = [
-            'title' => $track->name,
-            'artist' => $track->artists[0]->name,
-            'track_id' => $track->id,
-            'artist_id' => $track->artists[0]->id
-        ];
-
-        Song::store($song_data);
+        //this is zipping the arrays from the request parameters
+        $songs_data = array_map(null, request()->titles, request()->artist_names, request()->track_ids,  request()->artist_ids);
```

Comment:
Do you really need `array_map()` to generate the array? It might be enough to use square brackets:
```
$songs_data = [request()->titles, request()->artist_names, request()->track_ids,  request()->artist_ids];
```

Reply:
The point is to mix the arrays. So the first element of each array is put into an array, the second elements are put into another array... maybe the intuitive thing would be to loop through the arrays, but I like this method better